### PR TITLE
Use v-model for data browser selected rows

### LINKED ISSUE #172: Use v-model for Data Browser selected array.
Allow for 2-way binding for what rows in the browser are selected.  Lets external operations like `delete` update the selected array.

### PR DESCRIPTION
**Pull request by subdavis**
resolves #172 #179

This is actually necessary to address needs in #179.  Without this change, it isn't possible for downstream to change the value of selected.

### CODE REVIEW

**Comment by zachmullen** [comment]
Heh, I literally just ran into this 5 minutes ago.

**Comment by zachmullen** [comment]
To elaborate, a DANDI requirement is stated thus (emphasis mine):

> - when an item is selected from the search results list, the navigation should go to that items location in the hierarchy widget, **with that item's row highlighted**, and the details view for that item displayed

**Comment by subdavis** [comment]
Turns out the select behavior was completely broken since Vuetify 2.

Vuetify now provides the header with methods to toggleAll.

This still isn't working perfectly, but there are a ton of tricky edge cases where v-checkbox can get "backwards" because its props don't match its internal state.  The checkbox can look "on" when its `value` or `inputValue` are false because DOM checkbox state still exists.

File: src/components/DataBrowser.vue [hunk 1]
```diff
@@ -292,15 +296,15 @@ girder-data-table.girder-file-browser(
...
       th.pl-3.pr-0(width="1%", v-if="internalSelectable")
         v-checkbox.secondary--text.text--darken-1.pr-2(
             color="accent",
             hide-details,
-            :input-value="props.all",
-            :indeterminate="selected.length > 0 && !props.all",
-            @click.native="toggleAll")
+            :input-value="props.everyItem",
+            :indeterminate="internalValue.length > 0 && !props.everyItem",
+            @click.native="on['toggle-select-all'](!props.everyItem)")
```
Comment (subdavis) [inline src/components/DataBrowser.vue]: I'm confident I've done something real bad here, but I'm not sure what.

**Comment by subdavis** [comment]
As discussed in #172 the default behavior is now to clear the selection when the working directory changes.  When moving between pages of results within the same directory, the vuetify-builtin behavior of DataTable will be used.

The ability to select items across folders is no longer supported.  If we need to enable this ability later, we can.

**Pull request merged**